impeller/entity/entity_pass.cc

```diff
@@ -110,12 +110,39 @@ std::optional<Rect> EntityPass::GetElementsCoverage(
     if (auto entity = std::get_if<Entity>(&element)) {
       coverage = entity->GetCoverage();
 
+      // When the coverage limit is std::nullopt, that means there is no limit,
+      // as opposed to empty coverage.
       if (coverage.has_value() && coverage_limit.has_value()) {
-        coverage = coverage->Intersection(coverage_limit.value());
+        const auto* filter = entity->GetContents()->AsFilter();
+        if (!filter || !filter->HasBasisTransformations()) {
+          coverage = coverage->Intersection(coverage_limit.value());
+        }
       }
-    } else if (auto subpass =
+    } else if (auto subpass_ptr =
                    std::get_if<std::unique_ptr<EntityPass>>(&element)) {
-      coverage = GetSubpassCoverage(*subpass->get(), coverage_limit);
+      auto& subpass = *subpass_ptr->get();
+
+      std::optional<Rect> unfiltered_coverage =
+          GetSubpassCoverage(subpass, std::nullopt);
+      if (!unfiltered_coverage.has_value()) {
+        continue;
+      }
+
+      std::shared_ptr<FilterContents> image_filter =
+          subpass.delegate_->WithImageFilter(*unfiltered_coverage,
+                                             subpass.xformation_);
+      if (image_filter) {
+        Entity subpass_entity;
+        subpass_entity.SetTransformation(subpass.xformation_);
+        coverage = image_filter->GetCoverage(subpass_entity);
+      } else {
+        coverage = unfiltered_coverage;
+      }
+
+      if (coverage.has_value() && coverage_limit.has_value() &&
+          (!image_filter || !image_filter->HasBasisTransformations())) {
+        coverage = coverage->Intersection(coverage_limit.value());
+      }
```

Comment:
nit: this code feels complicated enough that you might want to pull it into a helper method.

Reply:
Done.